### PR TITLE
Fix validation of minHops and maxHops

### LINKED ISSUE #3060: Crash due to untrue assertion `e->minHops <= e->maxHops`
When running the following queries against an empty database:
```cypher
MERGE ()
-
MATCH ()<-[*..0]-() RETURN 0
```

The RedisGraph instance crashes because of the assertion `e->minHops <= e->maxHops` being untrue.

I encountered this issue when testing queries against **RedisGraph v2.12.0** compiled with address sanitization in a Docker container running **alpine v.3**.

<details>

<summary>Redis Bug Report</summary>

```
=== REDIS BUG REPORT START: Cut & paste starting from here ===
10:M 24 Apr 2023 08:58:45.448 # === ASSERTION FAILED ===
10:M 24 Apr 2023 08:58:45.448 # ==> /app/src/execution_plan/ops/op_cond_var_len_traverse.c:27 'e->minHops <= e->maxHops' is not true

------ STACK TRACE ------

Backtrace:
redis-server *:6379(_serverAssert+0x8c)[0x562e70194b8c]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(+0x69647c)[0x7fd0afd1747c]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(+0x693c00)[0x7fd0afd14c00]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(OpBase_Consume+0x44)[0x7fd0afd06494]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(+0x6c3abf)[0x7fd0afd44abf]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(OpBase_Consume+0x44)[0x7fd0afd06494]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(+0x6c53ad)[0x7fd0afd463ad]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(OpBase_Consume+0x44)[0x7fd0afd06494]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(ExecutionPlan_Execute+0x2f5)[0x7fd0afcf1c75]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(+0x6407ba)[0x7fd0afcc17ba]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(_query+0x533)[0x7fd0afcc0ce3]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(Graph_Query+0x19)[0x7fd0afcc30e9]
/app/bin/linux-x64-debug-asan/src/redisgraph.so(+0x831d8a)[0x7fd0afeb2d8a]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x7ea7)[0x7fd0b7bd3ea7]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f)[0x7fd0b7af1a2f]

------ INFO OUTPUT ------
# Server

redis_version:7.0.11

redis_git_sha1:00000000

redis_git_dirty:0

redis_build_id:5c712dc4cb9cfb70

redis_mode:standalone

os:Linux 6.2.10-arch1-1 x86_64

arch_bits:64

monotonic_clock:POSIX clock_gettime

multiplexing_api:epoll

atomicvar_api:c11-builtin

gcc_version:10.2.1

process_id:10

process_supervised:no

run_id:69f5886dcc1741064a6d89e25f9a8ff786217a10

tcp_port:6379

server_time_usec:1682326725447012

uptime_in_seconds:5

uptime_in_days:0

hz:10

configured_hz:10

lru_clock:4605125

executable:/redis/redis-server

config_file:

io_threads_active:0



# Clients

connected_clients:1

cluster_connections:0

maxclients:10000

client_recent_max_input_buffer:0

client_recent_max_output_buffer:0

blocked_clients:1

tracking_clients:0

clients_in_timeout_table:0



# Memory

used_memory:1787016

used_memory_human:1.70M

used_memory_rss:42557440

used_memory_rss_human:40.59M

used_memory_peak:1787016

used_memory_peak_human:1.70M

used_memory_peak_perc:129.57%

used_memory_overhead:928920

used_memory_startup:928664

used_memory_dataset:858096

used_memory_dataset_perc:99.97%

allocator_allocated:1223792

allocator_active:1409024

allocator_resident:4825088

total_system_memory:8039120896

total_system_memory_human:7.49G

used_memory_lua:31744

used_memory_vm_eval:31744

used_memory_lua_human:31.00K

used_memory_scripts_eval:0

number_of_cached_scripts:0

number_of_functions:0

number_of_libraries:0

used_memory_vm_functions:32768

used_memory_vm_total:64512

used_memory_vm_total_human:63.00K

used_memory_functions:184

used_memory_scripts:184

used_memory_scripts_human:184B

maxmemory:0

maxmemory_human:0B

maxmemory_policy:noeviction

allocator_frag_ratio:1.15

allocator_frag_bytes:185232

allocator_rss_ratio:3.42

allocator_rss_bytes:3416064

rss_overhead_ratio:8.82

rss_overhead_bytes:37732352

mem_fragmentation_ratio:45.82

mem_fragmentation_bytes:41628632

mem_not_counted_for_evict:0

mem_replication_backlog:0

mem_total_replication_buffers:0

mem_clients_slaves:0

mem_clients_normal:0

mem_cluster_links:0

mem_aof_buffer:0

mem_allocator:jemalloc-5.2.1

active_defrag_running:0

lazyfree_pending_objects:0

lazyfreed_objects:0



# Persistence

loading:0

async_loading:0

current_cow_peak:0

current_cow_size:0

current_cow_size_age:0

current_fork_perc:0.00

current_save_keys_processed:0

current_save_keys_total:0

rdb_changes_since_last_save:1

rdb_bgsave_in_progress:0

rdb_last_save_time:1682326720

rdb_last_bgsave_status:ok

rdb_last_bgsave_time_sec:-1

rdb_current_bgsave_time_sec:-1

rdb_saves:0

rdb_last_cow_size:0

rdb_last_load_keys_expired:0

rdb_last_load_keys_loaded:0

aof_enabled:0

aof_rewrite_in_progress:0

aof_rewrite_scheduled:0

aof_last_rewrite_time_sec:-1

aof_current_rewrite_time_sec:-1

aof_last_bgrewrite_status:ok

aof_rewrites:0

aof_rewrites_consecutive_failures:0

aof_last_write_status:ok

aof_last_cow_size:0

module_fork_in_progress:0

module_fork_last_cow_size:0



# Stats

total_connections_received:1

total_commands_processed:4

instantaneous_ops_per_sec:0

total_net_input_bytes:287

total_net_output_bytes:216

total_net_repl_input_bytes:0

total_net_repl_output_bytes:0

instantaneous_input_kbps:0.00

instantaneous_output_kbps:0.00

instantaneous_input_repl_kbps:0.00

instantaneous_output_repl_kbps:0.00

rejected_connections:0

sync_full:0

sync_partial_ok:0

sync_partial_err:0

expired_keys:0

expired_stale_perc:0.00

expired_time_cap_reached_count:0

expire_cycle_cpu_milliseconds:0

evicted_keys:0

evicted_clients:0

total_eviction_exceeded_time:0

current_eviction_exceeded_time:0

keyspace_hits:4

keyspace_misses:1

pubsub_channels:0

pubsub_patterns:0

pubsubshard_channels:0

latest_fork_usec:0

total_forks:0

migrate_cached_sockets:0

slave_expires_tracked_keys:0

active_defrag_hits:0

active_defrag_misses:0

active_defrag_key_hits:0

active_defrag_key_misses:0

total_active_defrag_time:0

current_active_defrag_time:0

tracking_total_keys:0

tracking_total_items:0

tracking_total_prefixes:0

unexpected_error_replies:0

total_error_replies:0

dump_payload_sanitizations:0

total_reads_processed:4

total_writes_processed:3

io_threaded_reads_processed:0

io_threaded_writes_processed:0

reply_buffer_shrinks:0

reply_buffer_expands:0



# Replication

role:master

connected_slaves:0

master_failover_state:no-failover

master_replid:f472e61425ce2885535327e57a6b59b2920efdf1

master_replid2:0000000000000000000000000000000000000000

master_repl_offset:0

second_repl_offset:-1

repl_backlog_active:0

repl_backlog_size:1048576

repl_backlog_first_byte_offset:0

repl_backlog_histlen:0



# CPU

used_cpu_sys:0.023204

used_cpu_user:0.022746

used_cpu_sys_children:0.000000

used_cpu_user_children:0.000000

used_cpu_sys_main_thread:0.000000

used_cpu_user_main_thread:0.004214



# Modules

module:name=graph,ver=21200,api=1,filters=0,usedby=[],using=[],options=[]



# Commandstats

cmdstat_ping:calls=1,usec=1,usec_per_call=1.00,rejected_calls=0,failed_calls=0

cmdstat_graph.QUERY:calls=3,usec=2822,usec_per_call=940.67,rejected_calls=0,failed_calls=0



# Errorstats



# Latencystats

latency_percentiles_usec_ping:p50=1.003,p99=1.003,p99.9=1.003

latency_percentiles_usec_graph.QUERY:p50=1179.647,p99=1523.711,p99.9=1523.711



# Cluster

cluster_enabled:0



# Keyspace

db0:keys=1,expires=0,avg_ttl=0


------ CLIENT LIST OUTPUT ------
id=6 addr=172.17.0.1:45720 laddr=172.17.0.2:6379 fd=8 name= age=0 idle=0 flags=b db=0 sub=0 psub=0 ssub=0 multi=-1 qbuf=0 qbuf-free=20474 argv-mem=79 multi-mem=0 rbs=16384 rbp=16384 obl=0 oll=0 omem=0 tot-mem=37743 events=r cmd=graph.QUERY user=default redir=-1 resp=2

------ MODULES INFO OUTPUT ------
# graph_executing commands

graph_command:GRAPH.QUERY CYPHER TIMEOUT_DEFAULT="30000" MATCH ()<-[*..0]-() RETURN 0


------ CONFIG DEBUG OUTPUT ------
client-query-buffer-limit 1gb
activedefrag no
lazyfree-lazy-user-del no
lazyfree-lazy-eviction no
repl-diskless-sync yes
slave-read-only yes
list-compress-depth 0
lazyfree-lazy-user-flush no
lazyfree-lazy-server-del no
proto-max-bulk-len 512mb
lazyfree-lazy-expire no
replica-read-only yes
repl-diskless-load disabled
io-threads-do-reads no
sanitize-dump-payload no
io-threads 1

------ FAST MEMORY TEST ------
10:M 24 Apr 2023 08:58:45.451 # main thread terminated
10:M 24 Apr 2023 08:58:45.452 # Bio thread for job type #0 terminated
10:M 24 Apr 2023 08:58:45.452 # Bio thread for job type #1 terminated
10:M 24 Apr 2023 08:58:45.452 # Bio thread for job type #2 terminated

Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.

=== REDIS BUG REPORT END. Make sure to include from START to END. ===
```

</details>

### Steps to reproduce
Run the following queries against an empty database and observe the database crashes:
```cypher
MERGE ()
-
MATCH ()<-[*..0]-() RETURN 0
```

### Expected behavior
The query should run successfully

### Actual behavior
The database crashes, with the untrue assertion: `e->minHops <= e->maxHops`

### PR DESCRIPTION
**Pull request by nafraf**
This patch is to fix #3060 

If ```minHops > maxHops``` the corresponding error message is returned.

### CODE REVIEW

**Comment by filipecosta90** [comment]
### Automated performance analysis summary

This comment was automatically generated given there is performance data available.

In summary:
- Detected a total of 7 stable tests between versions.
- Detected a total of 1 highly unstable benchmarks.
- Detected a total of 2 regressions bellow the regression water line 5.0.

You can check a comparison in detail via the [grafana link](https://benchmarksrediscom.grafana.net/d//SH9_rQYGz/?var-branch=master&var-branch=fix-min-maxHops-check)

### Comparison between master and fix-min-maxHops-check.

Time Period from 30 days ago. (environment used: oss-standalone)

|                         Test Case                          |Baseline master (median obs. +- std.dev)|Comparison fix-min-maxHops-check (median obs. +- std.dev)|% change (higher-better)|            Note             |
|------------------------------------------------------------|----------------------------------------|---------------------------------------------------------|------------------------|-----------------------------|
|ENTITY_COUNT                                                | 33324 +- 0.0%  (7 datapoints)          | 33323 +- nan%  (1 datapoints)                           |-0.0%                   |-- no change --              |
|EXPLICIT-EDGE-DELETION                                      | 100 +- 0.0%  (7 datapoints)            | 100 +- nan%  (1 datapoints)                             |0.0%                    |-- no change --              |
|GRAPH500-SCALE_18-EF_16-1_HOP                               | 33325 +- 24.4% UNSTABLE (7 datapoints) | 33326 +- nan%  (1 datapoints)                           |0.0%                    |UNSTABLE (very high variance)|
|GRAPH500-SCALE_18-EF_16-2_HOP_MIXED_READ_75perc_WRITE_25perc| 56 +- 2.9%  (7 datapoints)             | 56 +- nan%  (1 datapoints)                              |0.0%                    |-- no change --              |
|IMPLICIT-EDGE-DELETION                                      | 17 +- 0.0%  (7 datapoints)             | 17 +- nan%  (1 datapoints)                              |-0.0%                   |-- no change --              |
|INSPECT-EDGES                                               | 2500 +- 0.0%  (7 datapoints)           | 2500 +- nan%  (1 datapoints)                            |0.0%                    |-- no change --              |
|MIX_READ_WRITE                                              | 10000 +- 0.0%  (7 datapoints)          | 9090 +- nan%  (1 datapoints)                            |-9.1%                   |REGRESSION                   |
|NODE-INDEX-LOOKUP                                           | 33 +- 0.0%  (7 datapoints)             | 33 +- nan%  (1 datapoints)                              |0.0%                    |-- no change --              |
|SORT_ENTITIES                                               | 100 +- 0.0%  (7 datapoints)            | 100 +- nan%  (1 datapoints)                             |0.0%                    |-- no change --              |
|allShortestPaths-10hop-100Kpaths                            | 30 +- 4.4%  (7 datapoints)             | 27 +- nan%  (1 datapoints)                              |-9.1%                   |REGRESSION                   |

File: tests/flow/test_relation_patterns.py [hunk 1]
```diff
@@ -316,3 +328,52 @@ def test11_shared_node_detection(self):
...
+        g.query(q)
+
+        query_to_expected_result = {
+            "MATCH (a)-[r*2..2]->(b) RETURN size(nodes(r))" : [[3]],
+            "MATCH (a)-[r:R*2..2]->(b) RETURN size(nodes(r))" : [[3]],
+            "MATCH (a)-[r*1..2]->(b) RETURN size(nodes(r)) AS x ORDER BY x" : [[2], [2], [3]],
+            "MATCH (a)-[r*0..2]->(b) RETURN size(nodes(r)) AS x ORDER BY x" : [[1], [1], [1], [2], [2], [3]],
+            "MATCH (a)-[r*0..1]->(b) RETURN size(nodes(r)) AS x ORDER BY x" : [[1], [1], [1], [2], [2]],
+            "MATCH (a)-[r*0..0]->(b) RETURN size(nodes(r)) AS x ORDER BY x" : [[1], [1], [1]],
+        }
```
Comment (swilly22) [inline tests/flow/test_relation_patterns.py:362]: consider switching to tuples:
```python
query_to_expected_result = [
   ("MATCH (a)-[r*2..2]->(b) RETURN size(nodes(r))", [[3]]),
   ...
]
...

for query, expected_result in query_to_expected_result
...
```

**Review by swilly22 — CHANGES_REQUESTED** [review]
Looks good, a few comments

File: tests/flow/test_relation_patterns.py (hunk 1 above)
Comment (nafraf) [reply tests/flow/test_relation_patterns.py]: Done.

File: tests/flow/test_relation_patterns.py [hunk 2]
```diff
@@ -316,3 +328,52 @@ def test11_shared_node_detection(self):
         result = g.query(q)
         self.env.assertEquals(result.nodes_created, 0)
         self.env.assertEquals(result.relationships_created, 0)
+
+    # Test patterns with length less than zero
+    def test12_lt_zero_hop_traversals(self):
```
[the remark this answers is not in the document]
Comment (nafraf) [reply tests/flow/test_relation_patterns.py]: OK. Done in [d187be0](https://github.com/RedisGraph/RedisGraph/pull/3076/commits/d187be0cb04dbbee7489c6a35e92429321d2a8b1)

File: tests/flow/test_relation_patterns.py [hunk 3]
```diff
@@ -51,9 +58,14 @@ def test01_one_hop_traversals(self):
         query = """MATCH (a)-[*1]->(b) RETURN a.val, b.val ORDER BY a.val, b.val"""
         result_d = redis_graph.query(query)
 
+        # Conditional traversal with minimum length by default
```
[the remark this answers is not in the document]
Comment (nafraf) [reply tests/flow/test_relation_patterns.py]: OK. Change done in [d187be0](https://github.com/RedisGraph/RedisGraph/pull/3076/commits/d187be0cb04dbbee7489c6a35e92429321d2a8b1)

**Pull request merged**